pmaudit

```diff
@@ -490,6 +490,17 @@ def main():
         '-d', '--depsfile',
         metavar='FILE',
         help="save audit data in makefile format to FILE")
+    parser.add_argument(
+        '--shell',
```

Comment:
I'm happy to use style checkers.  I use pylint elsewhere, and also using pycodestyle is fine.

I'm less excited to run "mypystyle".  I'd far rather directly run widely-tools like pylint and pycodestyle (and check in the local style for that project as part of the project).  But I'll do it.  That said, I encourage you to convince someone to add your style preferences to widely-used tools instead of doing it yourself.

Since mypystyle doesn't have an installer or install instructions, I had hand-jam its installation.  It seems to run. I did something like this (I already had pylint installed):
~~~~
git clone https://github.com/boyski/mypystyle
pip3 install pycodestyle
cd pmaudit/
rsync -a ~/mypystyle/etc/ ./etc/
PATH="$PATH:$HOME/mypystyle/bin"
mypystyle pmaudit
~~~~

This produced the following pylint error report (and no others);

~~~~
pmaudit.py:381:15: E1123: Unexpected keyword argument 'encoding' in constructor call (unexpected-keyword-arg)
~~~~

Where line 381 is:
~~~~
        proc = subprocess.Popen(cmd, stdin=subprocess.PIPE, encoding='utf-8')
~~~~

The message is correct but misleading.  I'm on Ubuntu 16 with Python 3.5, where Popen does not have "encoding=...".  Not obvious what can be done there, so I'm ignoring the message.

Reply:
I only used SHELL because the original code worked that way.   I will dump that nonsense, which makes it simpler: SHELL will simply default to /bin/sh, and if you want something else you pass --shell new_shell.

Reply:
Now that I'm not trying to support the SHELL environment variable, it's easy:

~~~~
    parser.add_argument(
        '--shell', default='/bin/sh',
        help="name of shell to run (default=%(default)s)")
~~~~